src/_guides/language/language-tour.md

```diff
@@ -4241,6 +4240,33 @@ Keep in mind the following when you use deferred loading:
   using <code>deferred as <em>namespace</em></code>.
   The `loadLibrary()` function returns a [`Future`](/guides/libraries/library-tour#future).
 
+#### The `library` directive
+
+The only reason to use a library directive is
+to attach library-level [doc comments][] or [metadata annotations][].
+Library-level comments and annotations need to prepend
+a `library` declaration at the start of a file.
+There's also no reason to name a library using the directive;
+in fact, it's [discouraged][].
+
+{:.bad}
+<?code-excerpt "misc/lib/effective_dart/usage_bad.dart (library-dir)"?>
+{% prettify dart tag=pre+code %}
+
+library my_library;
+{% endprettify %}
+
+{:.good}
+<?code-excerpt "misc/lib/effective_dart/docs_good.dart (library-doc)"?>
+{% prettify dart tag=pre+code %}
+/// A really great test library.
+@TestOn('browser')
+library;
+{% endprettify %}
```

Comment:
Could you talk more about your thinking on including this bad-v-good section in the language tour? Presumably, most people visiting the language tour will be learners, so they may not even know about the syntax with a name. 

I guess it just feels a bit weird to me, as this is a reference, not really about what's good or bad, hence why you have this documented in Effective Dart as well. We do outline a few recommendations and link to Effective Dart, but the alternative to those recommendations still have a reason.  

Just would like to hear your thoughts :)

Reply:
Oh! That was not a style choice, I just cut it from the section I added to Effective Dart at first 😅 
_Maybe_ I thought there were other good/bad snippets in the language tour....but whatever the reason was not intentional. Thanks for catching that!